evcouplings/align/protocol.py

```diff
@@ -38,6 +40,106 @@
     add_full_header
 )
 
+# define the gap threshold for inclusion in HMM's build by HMMbuild. 
+SYMFRAC_HMMBUILD = 0.0
+
+
+def _make_hmmsearch_raw_fasta(ar, prefix):
+    """
+    HMMsearch results do not contain the query sequence
+    so we must construct a raw_fasta file with the query 
+    sequence as the first hit, to ensure proper numbering. 
+    The search result is filtered to only contain the columns with
+    match states to the HMM, which has a one to one mapping to the
+    query sequence.
+
+    Paramters
+    ---------
+    ar: dict
+        Alignment result dictionary, output by run_hmmsearch
+    prefix: str
+        Prefix for file creation
+
+    Returns
+    -------
+    ali: EVcouplings.align.Alignment
+        Alignment object
+
+    """
+    def _add_gaps_to_query(query_sequence_ali, ali):
+
+        new_sequence = ""
+        seq = list(query_sequence_ali .matrix[0, :])
+
+        # loop through every position in the HMMsearch hits
+        for i in range(0, len(ali.annotation["GC"]["RF"])):
```

Comment:
1) would be safer to check here if annotation actually exists and to raise a meaningful exception if not

2) range(0, ..) should just be range(...)

Reply:
We already checked if the annotation exists in lines 105-109, and raise a ValueError if not